sdk/go/internal/engineconn/bin/bin.go

```diff
@@ -0,0 +1,179 @@
+package bin
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"dagger.io/dagger/internal/engineconn"
+)
+
+func init() {
+	engineconn.Register("bin", New)
+}
+
+func New(u *url.URL) (engineconn.EngineConn, error) {
+	path := u.Host + u.Path
```

Comment:
maybe nit: if you stop right here and don't do the ifs below, then `bin://dagger-engine-session` would work out of the box (cmd exec will look in PATH if it's not a relative/absolute path)

We'd lose early validation though of the binary being executable

Reply:
Yeah good call. I left in some early checks but fixed it up so that we can support `bin://dagger-engine-session`; did the same in Python too. NodeJS will support that now too afaict but without the early checks, which is fine I think.